lib/chef/azure/service.rb

```diff
@@ -6,100 +6,76 @@ class ChefService
   include ChefAzure::Shared
   AZURE_CHEF_SERVICE_PID_FILE = "azure-chef-client.pid"
   AZURE_CHEF_CRON_NAME = 'azure_chef_extension'
+  DEFAULT_CHEF_SERVICE_INTERVAL = 30
+  CLIENT_RB_INTERVAL_ATTRIBUTE_NAME = 'interval'
 
-  # TODO - make these methods idempotent
-  def install(log_location)

```

Comment:
@btm `install` method was written previously to install `chef-service` on `Windows` platform, while another method `enable` was their to install `chef-service cronjob` on `Linux` platform. However, these two methods `install` and `enable` have been combined now. As per the `chef-service` current status and transitions in `chef_service_interval` option, this method will take care to take the appropriate action.